catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/reducer.js

```diff
@@ -45,7 +47,20 @@ export const fetch = () => (dispatch) => {
     .then((res) => res.json())
     .then((json) => {
       const config = fromJS(json).getIn(configPath)
-        .update('imageryProviders', (providers) => fromJS(JSON.parse(providers || '[]')))
+        .update('imageryProviders', (providers) => {
+          try {
+            const parsed = JSON.parse(providers)
+            const err = validateStructure(parsed)
+            if (err !== undefined) {
+              throw Error(err)
```

Comment:
❓ Throwing this error here and catching it on L58 feels like using the error as control flow. Instead of having the `if (err != undefined)` check here, could we have `validateStructure()` throw the `Error` instead and never return `undefined`?